lib/pbench/cli/agent/commands/results/__init__.py

```diff
@@ -1,94 +1,173 @@
+                sys.exit(1)
+
+        results_webserver = self.config.get("results", "webserver")
+        if not results_webserver:
+            logger.error(
+                "No web server host configured from which we can fetch the FQDN of the host to which we copy/move results"
+            )
+            logger.debug("'webserver' key in 'results' section not set")
+
+        server_rest_url = self.config.get("results", "server_rest_url")
+        response = requests.get(f"{server_rest_url}/host_info")
+        if response.status_code not in [200, 201]:
+            logger.error(
+                "Unable to determine results host info from %s/host_info",
+                server_rest_url,
+            )
+            sys.exit(1)
+        if response.text.startswith("MESSAGE"):
+            message = response.text.split("===")[1]
+            logger.info("*** Message from sysadmins of %s:", results_webserver)
+            logger.info("***\n*** %s", message)
+            logger.info("***\n*** No local actions taken.")
+            sys.exit(1)
+        results_path_prefix = response.text.split(":")[1]
+        if not results_path_prefix:
+            logger.error(
+                "fetch results host info did not contain a path prefix: %s",
+                response.text,
+            )
             sys.exit(1)
 
         try:
-            shutil.rmtree(result_dir)
-        except OSError:
-            logger.error("rm failed to remove the %s directory hierarchy", result_dir)
+            temp_dir = tempfile.mkdtemp(
+                dir=self.config.pbench_tmp, prefix="pbench-move-results."
+            )
+        except Exception:
+            logger.error("Failed to create temporary directory")
             sys.exit(1)
 
-        runs_copied += 1
+        user = self.context.user if self.context.user else "pbench"
+        prefix = self.context.prefix if self.context.prefix else ""
+
+        runs_copied = 0
+        failures = 0
 
-    if runs_copied + failures > 0:
-        logger.debug(
-            "successfully moved %s runs, encountered %s failures", runs_copied, failures
-        )
+        for dirent in config.pbench_run.iterdir():
+            if not self.dirent.is_dir():
+                continue
+            if dirent.name.startswith("tools-") or dirent.name == "tmp":
+                continue
+            result_dir = dirent
+            mrt = MakeResultTb(result_dir, temp_dir, user, prefix, self.config, logger)
+            make_return_val = mrt.check_result_target_dir(result_dir, temp_dir)
+            if make_return_val:
+                sys.exit(1)
+            status, result_tb_name = mrt.make_result_tb()
+
+            if status == 0:
+                if os.path.exists(result_tb_name):
+                    assert (
+                        result_tb_name
+                    ), "Logic bomb!  make_result_tb() always returns a tar ball name"
```

Comment:
How could `result_tb_name` be `None` if it's a filepath that `.exists()`???

Reply:
I think I'm in violent agreement with you, Dave.  So, the question is, is it OK to have in the code "logic bomb" assertions (which the developer honestly thinks can never happen --otherwise, they wouldn't refer to themselves as a "logic bomb"...).

I would assert (heh...) that there are four cases:

1. The developer thinks that a specific problem might occur, so s/he puts an `if` test in the code which handles it.
2. The developer thinks that a specific problem _might_ occur, but _shouldn't_ under reasonable conditions, so s/he puts an `assert` test in the code which handles it (ungracefully).
3. The developer thinks that a specific problem should _never_ occur, but is nevertheless concerned that it might, so s/he puts in a "logic bomb" to indicate that the situation violated the implementation presumptions.  This is particularly appropriate if it prevents execution from proceeding to an operation which is particularly expensive, risky, or likely to obscure the problem which reported by the assertion.
4. The developer is confident that a specific problem will _never_ occur, so s/he omits any test for it.

In this particular case, I'm on-board with removing this assertion...but I'm not in a hurry to do so because apparently someone thought it was important enough to insert it.  (But, yeah, if it's "just debugging code", then, yeah, it should be removed before this PR is moved out of "draft mode".)

Reply:
I added the logic bomb: https://github.com/distributed-system-analysis/pbench/commit/6be975825c6aac3aa052643e6602f2e5b47b2ed0

Reply:
PR #1947.

Reply:
The original intend of the assertion was for the previous context: `mrt.make_result_tb()` returned a tar ball.

I'd invert this `if` test and text for failure first, `continue`ing early, then continue the flow for success starting with an assertion, and dropping the `exists()` test.